File: tests/bindings/e2e/cpp/registration_restart_fixture.cpp
#include "common/descriptors.hpp"
#include "common/memory/backend.hpp"
#include "common/runtime/lease_store.hpp"
#include "common/runtime/queue_store.hpp"
#include "common/runtime/workflow_store.hpp"

int main()
{
    statespec::backend::memory::InMemoryBackend backend;

    {
        statespec::backend::runtime::RuntimeQueueStore queues;
        statespec::backend::runtime::RuntimeLeaseStore leases;
        statespec::backend::runtime::RuntimeWorkflowStore workflows;
        statespec_generated::bootstrap_runtime_catalog(backend, queues, leases, workflows);
    }

    {
        statespec::backend::runtime::RuntimeQueueStore queues;
        statespec::backend::runtime::RuntimeLeaseStore leases;
        statespec::backend::runtime::RuntimeWorkflowStore workflows;
        statespec_generated::bootstrap_runtime_catalog(backend, queues, leases, workflows);
    }
}

File: tests/bindings/e2e/java/RegistrationRestartFixture.java
package com.statespec.generated;

import com.statespec.backend.memory.InMemoryBackend;
import com.statespec.backend.runtime.LeaseStore;
import com.statespec.backend.runtime.QueueStore;
import com.statespec.backend.runtime.WorkflowStore;
import com.statespec.generated.runtime.RuntimeRegistration;

public final class RegistrationRestartFixture
{
    private RegistrationRestartFixture() {}

    public static void main(String[] args) throws Exception
    {
        var backend = new InMemoryBackend();

        RuntimeRegistration.bootstrapRuntimeCatalog(
            backend, new QueueStore(), new LeaseStore(), new WorkflowStore()
        );
        RuntimeRegistration.bootstrapRuntimeCatalog(
            backend, new QueueStore(), new LeaseStore(), new WorkflowStore()
        );
    }
}
